### ISSUE TITLE
keras->onnx, the batchnorm value loss

### ISSUE THREAD

**Issue by CristXu** [onnx/onnxmltools#323]
hi, i use this tool to convert a keras model into the onnx, but it seems that the value of the batchnorm layer is not the same as the value in the keras model, but the others value is right.
by the way, i use onnx_model = onnxmltools.convert_keras(keras_model, target_opset=7) to convert. so could anybody help me about that? why does this issue happen?

**Comment by jiafatom** [comment]
Can you try target opset=10? also you can try keras converter directly [here](https://github.com/onnx/keras-onnx) using opset=10. If it does not work, please share the model.

**Comment by CristXu** [comment]
Hi，
       Thanks for your response, because the onnxmltools used to convert the keras model, it will normalize the value in the batchnorm layer: mean, scale, var, bias; but it is the same as the before one and will take no effect on the converted model.
       By the way, the raw-data is params[0], params[1], params[2], params[3]; and the value after is gamma, beta, 0, 1
[cid:image001.png@01D53C92.8A1B62C0]

Regards，
       Crist

**Comment by jiafatom** [comment]
sorry I don't understand your reply. Does your conversion work using opset 10? onnxmltools will call [keras2onnx](https://github.com/onnx/keras-onnx), so can you try it directly? If it does not work, please share the model file.

**Comment by CristXu** [comment]
I mean that even though the value in the batchnorm layer of the onnx-model is not the same as the one in keras-model ( I convert the keras-model into the onnx-model with the onnxmodel ), because the convert-tool normalize the value in the batchnorm, make std to 1, and mean to 0, so I think if the convert-tool have some bugs with the keras-model before, but now, I figure out that the value is equal between the onnx-model & keras-model.
By the way, the convert tool has no bugs. It can works well.
About the normalize, you can find the detail in the keras2onnx/batch_norm.py , line 45.

**Comment by jiafatom** [comment]
Good to know that the model conversion works well. We do some transform on the value part because keras and onnx spec is different, so it is normal to see the value difference.
